fhir-server/src/main/java/au/csiro/pathling/fhirpath/operator/CombineOperator.java

```diff
@@ -57,7 +57,15 @@ public FhirPath invoke(@Nonnull final OperatorInput input) {
         Arrays.asList(operand.getIdColumn(), operand.getValueColumn()));
     if (operand instanceof NonLiteralPath) {
       final NonLiteralPath nonLiteralLeft = (NonLiteralPath) operand;
+      // If there is a this column, we need to preserve it.
       nonLiteralLeft.getThisColumn().ifPresent(columns::add);
+      // If the this context is a resource, we need to preserve all element columns.
+      input.getContext().getThisContext().ifPresent(thisContext -> {
+        if (thisContext instanceof ResourcePath) {
+          final ResourcePath thisResource = (ResourcePath) thisContext;
+          columns.addAll(thisResource.getElementColumns());
+        }
+      });
     }
```

Comment:
This looks a bit inelegant  with all the instance type checks. Maybe this should be a method on FhirPath? Either to trim or to produce the list of important columns for the path. 
Also I think some other path (UntypedResource ???) have some additional important columns (type I think).